backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/monitoringannotation/aspect/MonitorAspect.java

```diff
@@ -0,0 +1,135 @@
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.monitoringannotation.aspect;
+
+import de.metas.monitoring.adapter.MicrometerPerformanceMonitoringService;
+import de.metas.monitoring.adapter.PerformanceMonitoringService;
+import de.metas.monitoringannotation.annotation.Monitor;
+import de.metas.util.Services;
+import lombok.NonNull;
+import org.adempiere.ad.element.api.AdWindowId;
+import org.adempiere.ad.window.api.IADWindowDAO;
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.reflect.MethodSignature;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import org.springframework.web.context.request.RequestContextHolder;
+import org.springframework.web.context.request.ServletRequestAttributes;
+import org.springframework.web.servlet.HandlerMapping;
+
+import javax.servlet.http.HttpServletRequest;
+import java.lang.reflect.Method;
+import java.util.Map;
+import java.util.concurrent.Callable;
+
+@Aspect
+@Component
+public class MonitorAspect
+{
+	private final MicrometerPerformanceMonitoringService service;
+
+	public MonitorAspect(@NonNull final MicrometerPerformanceMonitoringService service)
+	{
+		this.service = service;
+	}
+
+	@Value("${performance.monitoring.enable:false}")
+	private Boolean perfMonEnvVar;
+
+	@Around("execution(* *(..)) && @annotation(de.metas.monitoringannotation.annotation.Monitor)")
+	public Object monitorMethod(ProceedingJoinPoint pjp) throws Throwable
+	{
+		final PerformanceMonitoringService.Metadata metadata;
+		final Callable callable = getCallableFromProceedingJoinPoint( pjp );
+
+		if(perfMonEnvVar = false)
+		{
+			callable.call();
+		}
+
+		Method method = ((MethodSignature) pjp.getSignature()).getMethod();
+		Monitor monitorAnnotation = method.getAnnotation(Monitor.class);
+
+		if(monitorAnnotation.type() == PerformanceMonitoringService.Type.REST_CONTROLLER_WITH_WINDOW_ID)
+		{
+			HttpServletRequest request = ((ServletRequestAttributes) RequestContextHolder.currentRequestAttributes()).getRequest();
+			Map pathVariables = (Map) request.getAttribute(HandlerMapping.URI_TEMPLATE_VARIABLES_ATTRIBUTE);
+			String windowId = (String)pathVariables.get("windowId");
+			String window = (Services.get(IADWindowDAO.class).retrieveWindowName(AdWindowId.ofRepoId(Integer.parseInt(windowId)))).getDefaultValue();
```

Comment:
extract IADWindowDAO service as class field.
extact this code snippet to `String getWindowName(HttpServletRequest request)` method.